### PR TITLE
Updates to Go IPFS 0.6.0

### LINKED ISSUE #353: Update docs to have Go-IPFS 0.6 be the default installation option.
We have several locations where we instruct the reader to install Go-IPFS 0.5.1. When Go-IPFS 0.6 is released, these locations need updating.

Deferred until 0.6 released. It's currently in [RC7](https://dist.ipfs.io/go-ipfs/v0.6.0-rc7).

### PR DESCRIPTION
**Pull request by johnnymatthews**
Closes #353.

### CODE REVIEW

**Comment by johnnymatthews** [comment]
Saw that 0.6 got released on Friday evening, so threw this together based off the [Go-IPFS 0.6 releases page](https://github.com/ipfs/go-ipfs/releases/tag/v0.6.0). Haven't seen any social posts about this release, so not sure if we're keeping it quiet until next week.

@momack2, this is pretty much a straight copy-and-paste from the [Go-IPFS 0.6 releases page](https://github.com/ipfs/go-ipfs/releases/tag/v0.6.0). [There's a preview available here.](https://bafybeidt2hjtpigxc2fb5le7e3zcye5w5koqnhlhtbpiootfgven76hadi.ipfs.dweb.link/recent-releases/go-ipfs-0-6/). This can merge when ready, not sure if we're keeping this release quiet for a reason.

**Comment by momack2** [comment]
Nope! I assume @aschmahmann just hasn't had time to put together a release blog post yet. =]

File: docs/recent-releases/go-ipfs-0-6/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+- QUIC by default.
+- Libp2p Noise as a new security transport.
+- Custom 404 pages and base36 support.
+- Gossipsub upgrade.
+- New peering subsystem.
+
+## Breaking changes
+
+There are no breaking changes in this release. However, there is a migration will run, and the QUIC transport has changed protocol versions.
```
Comment (momack2) [inline docs/recent-releases/go-ipfs-0-6/README.md:23]: this is grammatically incorrect ("there is a migration THAT will run", or "a migration will run"). Probably useful to indicate when and for what the migration will run

File: docs/recent-releases/go-ipfs-0-6/changelog.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
+---
+title: Changelog
```
Comment (momack2) [inline docs/recent-releases/go-ipfs-0-6/changelog.md:2]: I don't think we should include the changelog in the docs

File: docs/recent-releases/go-ipfs-0-6/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Features
+
+There are plenty of features packed into this release:
+
+- QUIC by default.
+- Libp2p Noise as a new security transport.
+- Custom 404 pages and base36 support.
+- Gossipsub upgrade.
+- New peering subsystem.
```
Comment (momack2) [inline docs/recent-releases/go-ipfs-0-6/README.md:19]: New peering command for persistent connections between libp2p nodes

File: docs/recent-releases/go-ipfs-0-6/changelog.md (hunk 2 above)
Comment (johnnymatthews) [reply docs/recent-releases/go-ipfs-0-6/changelog.md:2]: Fair enough. I'll just link to the changelog section on the release page.

File: docs/recent-releases/go-ipfs-0-6/README.md (hunk 3 above)
Comment (johnnymatthews) [reply docs/recent-releases/go-ipfs-0-6/README.md:19]: > New peering command for persistent connections between libp2p nodes.


Is this in addition to `New peering subsystem`, or does it replace that line?

**Comment by momack2** [comment]
Replace that line :)

**Pull request merged**